src/schnetpack/interfaces/ase_interface.py

```diff
@@ -271,54 +283,198 @@ def _load_model(self, model_file: str) -> schnetpack.model.AtomisticModel:
+                raise AtomsConverterError(
+                    "'{:s}' is not a property of your model. Please "
+                    "check the model "
+                    "properties!".format(prop)
+                )
+
+        self.results = results
+        self.model_results = model_results
+
     def calculate(
         self,
         atoms: ase.Atoms = None,
         properties: List[str] = ["energy"],
         system_changes: List[str] = all_changes,
     ):
         """
+        Calculate the properties for the given atoms.
+
         Args:
-            atoms (ase.Atoms): ASE atoms object.
-            properties (list of str): select properties computed and stored to results.
+            atoms (ase.Atoms or list): ASE atoms object.
+            properties (list of str): Properties to compute.
             system_changes (list of str): List of changes for ASE.
+
+        Returns:
+            None
+
         """
-        # First call original calculator to set atoms attribute
-        # (see https://wiki.fysik.dtu.dk/ase/_modules/ase/calculators/calculator.html#Calculator)
 
-        if self.calculation_required(atoms, properties):
+        if self.calculation_required(atoms=atoms, properties=properties):
+            # First call original calculator to set atoms attribute
+            # (see https://wiki.fysik.dtu.dk/ase/_modules/ase/calculators/calculator.html#Calculator)
             Calculator.calculate(self, atoms)
+            self._collect_results(atoms=atoms, properties=properties)
+            self._ase_specific_output_format(properties=properties)
 
-            # Convert to schnetpack input format
-            model_inputs = self.converter(atoms)
-            model_results = self.model(model_inputs)
 
-            results = {}
-            # TODO: use index information to slice everything properly
+class SpkEnsembleCalculator(SpkCalculator):
+    """
+    Calculator for neural network models for ensemble calculations.
+    Requires multiple models
+    """
+    # TODO: maybe calculate function should calculate all properties always
+
+    def __init__(
+            self,
+            model: Union[List[str], List[nn.Module]],
+            neighbor_list: schnetpack.transform.Transform,
+            energy_key: str = "energy",
+            force_key: str = "forces",
+            stress_key: Optional[str] = None,
+            energy_unit: Union[str, float] = "kcal/mol",
+            position_unit: Union[str, float] = "Angstrom",
+            device: Union[str, torch.device] = "cpu",
+            dtype: torch.dtype = torch.float32,
+            converter: callable = AtomsConverter,
+            transforms: Union[
+                schnetpack.transform.Transform, List[schnetpack.transform.Transform]
+            ] = None,
+            additional_inputs: Dict[str, torch.Tensor] = None,
+            auxiliary_output_modules: Optional[List] = None,
+            ensemble_average_strategy: Optional[EnsembleAverageStrategy] = None,
```

Comment:
we could use EnsembleAverageStrategy as a default and remove a code block in _collect_results